src/main/scala/backends/futil/FutilBackend.scala

```diff
@@ -194,6 +185,101 @@ private class FutilBackendHelper {
     List(mem)
   }
 
+  /** Returns the name of `p`, if it has one. */
+  def getPortName(p: Port) : String = {
+    p match {
+      case CompPort(id, _) => id.name
+      case ThisPort(id) => id.name
+      case HolePort(id, _) => id.name
+      case ConstantPort(_, _) => throw Impossible("Constant Ports do not have names.")
+    }
+  }
+
+    /** Returns a list of tuples (name, width) for each address port
+        in a memory. For example, a D1 Memory declared as (32, 1, 1)
+        would return List[("addr0", 1)].
+    */
+    def getAddrPortToWidths(typ: TArray, id: Id): List[(String, Int)] = {
```

Comment:
The Pythonic approach to this is:
```
    dims = dims2id[id] # Mapping from number of dims to compId, e.g. 1 : 'std_mem_d1'
    addresses = range(0, dims)
    indices = range(dims + 1, dims << 1 + 1)
    return [(f'addr{i}', args[n]) for (i, n) in zip(addresses, indices)]
```

so we should be able to shorten this here as well.